[CC-15601] AC Doc Documentation Support multiple abstract products as Promotional Products (#738)
spryker/spryker-docs: docs/scos/user/back-office-user-guides/202200.0/merchandising/discount/creating-cart-rules.md

    link: docs/scos/user/back-office-user-guides/page.version/merchandising/discount/managing-discounts.html
---

This guide describes how to create a *cart rule*.

A cart rule is a discount that is applied automatically when all attached discount conditions are fulfilled and if the cart rule is active. Unlike a voucher code, it does not require any input from the customer.

## Prerequisites

To start working with discounts, go to **Merchandising&nbsp;<span aria-label="and then">></span> Discount**.

> Review the reference information before you start, or just look up the necessary information as you go through the process.

Reviewer: I think it would be nice to link `reference information` to its section

## Creating cart rule discounts

To create a cart rule discount:
1. On the **Discount** page, in the top right corner, click **Create new discount**.
2. On the **Create new discount** page, in the **General Information** tab, do the following:
   1. In **STORE RELATION**, select stores you wish the discount to be active in.
   2. For **DISCOUNT TYPE**, select **Cart rule**.
   3. In the **NAME** field, specify the name for the discount.

>    4. Optional: in the **DESCRIPTION** field, enter a summary that explains the discount.

Reviewer: The first letter after `Optional:` should be capitalized. See [Optional steps](https://developers.google.com/style/procedures?hl=en#optional-steps).

Reviewer: Apply this in the other steps too.

   5. Optional: in the **PRIORITY** field, enter an integer value from `1` to `9999` for the discount priority.

>    6. Specify if the discount is either **NON-EXLUSIVE** or **EXCLUSINVE**.

Reviewer: ```suggestion
   6. Specify if the discount is **NON-EXLUSIVE** or **EXCLUSINVE**.
```

>    7. In **VALID FROM** and **VALID TO**, Specify the validity interval (lifetime) of the discount.

Reviewer: ```suggestion
   7. Select **VALID FROM** and **VALID TO** dates. 
```

Reviewer: This information is already in the reference

   8. To proceed to the **Discount calculation** tab, click **Next**.
3. On the **Discount calculation** tab, do the following:

>    1. In the **CALCULATOR TYPE** drop-down, select either **Percentage** or **Fixed amount**.

Reviewer: ```suggestion
   1. Select a **CALCULATOR TYPE**.
```

Reviewer: 1. Don't focus on the name of the page elements.
2. Mentioning the names of the drop-down items does not help with selecting. The reader will have to look up the reference anyway.

      {% info_block warningBox "Note" %}

      The next step varies based on the calculator type that you select:
      * **Fixed amount**: Enter the amounts (per currency and price mode, if applicable) to be discounted.
      * **Percentage**: Enter the percent value to be discounted.

>       {% endinfo_block %}

Reviewer: 2. Based on the **CALCULATOR TYPE** you've selected, do the following:
     * **Fixed amount**: Enter the amounts (per currency and price mode, if applicable) to be discounted.
     * **Percentage**: Enter the percent value to be discounted.

Reviewer: This is still a step, so it shouldn't be a note. 

Use present perfect to refer to the steps the reader performed as part of this procedure. 

In the conditional clause, directly refer to the configuration element that defines what the reader needs to do in the current step.

Author: Done

>    2. Select the **DISCOUNT APPLICATION TYPE** and define which products are eligible for the discount.

Reviewer: Separate the second part of this sentence and try to use the approach from the previous comment.

   3. To proceed to the **Conditions** tab, click **Next**.
4. On the **Conditions** tab, do the following:
    1. Select the **APPLY WHEN** conditions or click **Plain query** and enter the query manually.

>     2. Enter the value for **THE DISCOUNT CAN BE APPLIED IF THE QUERY APPLIES FOR AT LEAST X ITEM(S).** field.

Reviewer: ```suggestion
    2. For **THE DISCOUNT CAN BE APPLIED IF THE QUERY APPLIES FOR AT LEAST X ITEM(S).**, enter a numeric value.
```

5. To create the new discount, click **Save**.

### Reference information: Creating cart rule discounts

This section contains reference information you select and enter when creating cart rule discounts.

#### General information tab

The following table describes the attributes you enter and select on the **General information** tab:

<div class="width-100">

| ATTRIBUTE |DESCRIPTION  |
| --- | --- |
| STORE RELATION | Stores that your cart rule will be active in. You can select multiple stores.|

> | DISCOUNT TYPE | Drop-down list where you select either **Voucher codes** or **Cart rule** discount type. |

Reviewer: Explain what these discount types are. Give links to the respective sections in the feature overview.

Author: Done

| NAME | Unique name that will be displayed in the cart, in the **Available discounts** section on the Storefront, if the cart rule is valid. Should be short, but descriptive. |

> | DESCRIPTION | Explains the voucher and helps a customer understand why they are eligible for the discount and what they can receive. The description will be displayed in the **Cart calculation** section along with the applied discount amount if the discount is applicable to a product in the cart.|

Reviewer: ```suggestion
| DESCRIPTION | Describes the discount and helps a customer understand why they are eligible for the discount and what they can receive. The description will be displayed in the **Cart calculation** section along with the applied discount amount if the discount is applicable to a product in the cart.|
```

> | PRIORITY | Defines [the discount priority](/docs/scos/user/features/{{page.version}}/promotions-discounts-feature-overview.html#discount-priority). Represented as an integer value from `1` to `9999`, `1` being the highest priority and `9999` the lowest. |

Reviewer: Before providing a link, try to briefly explain what priority is.

Author: Done

| NON-EXCLUSIVE | Defines the discount exclusivity. Customers can redeem non-exclusive discounts in conjunction with other non-exclusive discounts.|
| EXCLUSIVE | Defines the discount exclusivity. When a discount is exclusive, no other discounts may be applied in conjunction. When a cart is eligible for multiple exclusive discounts, the discount with the highest value to the customer is applied. The exception to this is promotional product discounts. Query string discounts and promotional product discounts exclude only among each other. Promotional product discounts are not affected by exclusive query string discounts and conversely.|
| VALID FROM and VALID TO | Cart rule will be active between **VALID FROM** and **VALID TO** dates and times (in UTC), inclusive. For example, a discount applies to the cart starting from 01.12.2021 23:00 until 31.01.2022 22:59, UTC. |

</div>

{% info_block infoBox "Info" %}

> Name and describe discounts in a way that is meaningful for other Back Office users. Besides, the given name is displayed in the customer's cart. Therefore, it must be unique.

Reviewer: What value are we trying to add by this note?

{% endinfo_block %}

#### Discount calculation tab

This section contains information you need to know when working with discount calculations on the **Discount calculation** tab.

**CALCULATOR TYPE**

> <br>The discount can be calculated in two ways:

Reviewer: ```suggestion
A discount can be calculated in two ways:
```

> * **Percentage**: A discount is calculated as a percentage of the discounted items' prices. If selected, in the **VALUE** field, set the percentage value (for example, `25`).

Reviewer: Move the info from the parentheses to a separate sentence. See [Parentheses](https://developers.google.com/style/parentheses?hl=en).

Reviewer: Also, see [Examples](https://developers.google.com/style/format-examples).

Author: Done

> * **Fixed amount**: A fixed amount is discounted. If you select this type, specify the amount (**Gross price**, **Net price**, or both) for each currency used in your store.

Reviewer: Same as the previous.

Reviewer: Also, check the reset of the doc for parentheses.

Author: Done

You can find plain query examples in the following table.

|PLAIN QUERY|EXPLANATION|
|---|---|
|`day-of-week = '1'`|Discount applies if the order is placed on Monday.|
|`shipment-carrier != '1' AND price-mode = 'GROSS_MODE'` | Discount applies if the shipment carrier with the attribute `1` is not chosen, and gross pricing is selected.|
|`currency != 'EUR' OR price-mode = 'GROSS_MODE'` | Discount applies if the selected currency is not Euro, or the pricing mode is gross.|

{% info_block infoBox "Info" %}

For more information about tokens, on this page, see [Token description tables](#token-description-tables).

{% endinfo_block %}

**PROMOTIONAL PRODUCT**
<br>The **PROMOTIONAL PRODUCT** discount type lets you discount specific products at a fixed quantity when the discount conditions are met. A common use case is a "buy x, get y" discount, where a customer gets an item for free when they buy a certain product or spend a certain amount.

The following table describes attributes you enter if, for **DISCOUNT APPLICATION TYPE**, you select **PROMOTIONAL PRODUCT**:

| ATTRIBUTE | DESCRIPTION |
| --- | --- |

> | ABSTRACT PRODUCT SKU(S) | Stores comma-separated abstract product SKUs—products which will be promoted. |

Reviewer: ```suggestion
| ABSTRACT PRODUCT SKU(S) | Comma-separated abstract product SKUs — products that will be offered with a discount. |
```

> | MAXIMUM QUANTITY | Represents the limit on the number of units eligible for the discount. |

Reviewer: ```suggestion
| MAXIMUM QUANTITY | Maximum number of product units that can be discount. |
```

Reviewer: `Eligible` is more suitable when a product is qualified for a discount, not it's number.

Reviewer: Fix this and the previous one in all the reference information sections of discounts.

## Prerequisites

To start working with discounts, go to **Merchandising&nbsp;<span aria-label="and then">></span> Discount**.

## Creating a cart rule discount

To create a cart rule discount:
1. On the **Discount** page, in the top right corner, click **Create new discount**.
2. On the **Create new discount** page, in the **General Information** tab, do the following:

>    1. In **STORE RELATION**, select stores you wish the discount to be active in.

Reviewer: ```suggestion
   1. In **STORE RELATION**, select the stores you wish the discount to be active in.
```

Reviewer: Check _wish_ in https://developers.google.com/style/word-list?hl=en

2. On the **Create Discount** page, on the **General information** tab, do the following:
    1. In **STORE RELATION**, check the stores you wish the discount to be active in.
    2. From the **DISCOUNT TYPE** drop-down list, select a cart rule.

>     3. In the **NAME** field, specify the name for the discount. <br>It is displayed in the **Cart calculation** section.

Reviewer 2: ```suggestion
    3. In the **NAME** field, specify the name for the discount. <br>It is displayed in the **Cart calculation** section along with the applied discount amount.
```

>     4. _Optional_: in the **DESCRIPTION** field, enter the description of the discount. <br>The description is displayed in the cart, in the **Promotional products** section for products eligible for the discount.

Reviewer 2: ```suggestion
    4. _Optional_: in the **DESCRIPTION** field, enter a summary explaining the discount. <br>The description is displayed in the cart, in the **Promotional products** section (when applicable). Use wording that helps the customer understand why they are eligible for the discount and what they will receive.
```

>     4. _Optional_: in the **DESCRIPTION** field, enter a summary that explains the discount and helps the customer understand why they are eligible for the discount and what they can receive.<br>The description is displayed in the cart, in the **Promotional products** section (when applicable).

Author: @andriitserkovnyi better?
```suggestion
    4. _Optional_: in the **DESCRIPTION** field, enter a summary that explains the discount and helps the customer understand why they are eligible for the discount and what they can receive.<br>The description is displayed in the cart, in the **Promotional products** section when the discount is applicable
```
